recipes/polymorphic_value/all/conanfile.py

```diff
@@ -32,38 +32,42 @@ def _minimum_compilers_version(self):
         }
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        if self.info.settings.get_safe("compiler.cppstd"):
+            tools.build.check_min_cppstd(self, self._minimum_cpp_standard)
         min_version = self._minimum_compilers_version.get(
-            str(self.settings.compiler))
+            str(self.info.settings.compiler))
         if not min_version:
-            self.output.warn("{} recipe lacks information about the {} "
-                             "compiler support.".format(
-                                 self.name, self.settings.compiler))
+            self.output.warning("{} recipe lacks information about the {} "
+                                "compiler support.".format(
+                                    self.name, self.info.settings.compiler))
         else:
-            if tools.Version(self.settings.compiler.version) < min_version:
+            if tools.Version(self.info.settings.compiler.version) < min_version:
                 raise ConanInvalidConfiguration(
                     "{} requires C++{} support. "
                     "The current compiler {} {} does not support it.".format(
                         self.name, self._minimum_cpp_standard,
-                        self.settings.compiler,
-                        self.settings.compiler.version))
+                        self.info.settings.compiler,
+                        self.info.settings.compiler.version))
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
+
+    def layout(self):
+        basic_layout(self, src_folder="source")
```

Comment:
This is just a style convention, 😎 


```suggestion
        basic_layout(self, src_folder="src")
```

Reply:
You might want to have that in the docs then: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-layout-method

Reply:
@dvirtz Conan docs is different from ConanCenterIndex convention. It's well documented on ConanCenterIndex templates: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L70